frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts

```diff
@@ -346,7 +346,15 @@ type RowFilterBase<R> = {
   filterGroupName: string;
   type: string;
   items: RowFilterItem[];
+  filter?: (input: FilterValue, obj: R) => boolean;
+  defaultSelected?: string[];
+};
+
+export type RowNameLabelFilter<R = any> = {
```

Comment:
this type should have only `type` and `filter` props. Other have no meaning

Reply:
The problem is that when I added RowNameFilter to the RowFilter definition, there is code that depends on rilterGroupName and/or items and/or defaultSelected as part of the type definition.  There is code that is depending on items of RowFilter to have these items as part of the type definition.  By adding RowNameLabelFilter as an option of RowFilter, these options are not guaranteed to be part of the type definition and cause type errors.

See `/frontend/public/components/filter-toolbar.tsx`[ line 173](https://github.com/openshift/console/blob/master/frontend/public/components/filter-toolbar.tsx#L173) for an example.